interpreter/src/objects.rs

```diff
@@ -351,16 +358,17 @@ impl From<&Key> for Key {
 }
 
 // Convert Vec<T> to Value
-impl<T: Into<Value>> From<Vec<T>> for Value {
+impl<T: Into<Value> + 'static> From<Vec<T>> for Value {
     fn from(v: Vec<T>) -> Self {
-        Value::List(v.into_iter().map(|v| v.into()).collect::<Vec<_>>().into())
-    }
-}
-
-// Convert Vec<u8> to Value
-impl From<Vec<u8>> for Value {
-    fn from(v: Vec<u8>) -> Self {
-        Value::Bytes(v.into())
+        if std::any::TypeId::of::<Vec<T>>() == std::any::TypeId::of::<Vec<u8>>() {
+            Value::Bytes(Arc::new(unsafe {
+                // SAFETY: Checked Vec<T> is Vec<u8>, so they're the same type,
+                // just the compiler can't tell without specialization.
+                std::mem::transmute::<Vec<T>, Vec<u8>>(v)
+            }))
+        } else {
+            Value::List(v.into_iter().map(|v| v.into()).collect::<Vec<_>>().into())
+        }
     }
 }
```

Comment:
Basically merged `From<Vec<T>>` and `From<Vec<u8>>`, this was required to implement `From<u8>` for `Value`.

The if check should effectively be `if const == const` once `T` is resolved, so the compiler will optimize it away.

The only regression here is that now `T: 'static` but I can't think of a case where a Vec of non-static types would be used, so I think this is good enough for now because Value can't hold non-static types at the moment anyway.

Reply:
The life time is on instances of `T`s not the type `T`.
So this, which was perfectly fine before:

```rust
let string = "a".to_string();
let vec = vec![string.as_str()];
let v: Value = vec.into();
```

would not compile anymore:

```
974 |         let string = "a".to_string();
    |             ------ binding `string` declared here
975 |         let vec = vec![string.as_str()];
    |                   -----^^^^^^----------
    |                   |    |
    |                   |    borrowed value does not live long enough
    |                   cast requires that `string` is borrowed for `'static`
...
978 |     }
    |     - `string` dropped here while still borrowed
```

Reply:
It's on type `T`, so it requires that `&str` is `&'static str` because a value can't outlive its type.

But you're right, I added a test for it. It can be worked around though.

I probably should've feature gated serde in a separate PR, but what can you do... you know... \*looks away from separating that afterward\*